DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -101,23 +102,25 @@ internal class DataUploadWorker: DataUploadWorkerType {
                             case let .networkError(error: error):
                                 telemetry.error("Data upload finished with error", error: error)
                                 return
-                            case .none: continue
                             }
-                        } catch let error {
-                            // If upload can't be initiated do not retry, so drop the batch:
-                            self.fileReader.markBatchAsRead(batch, reason: .invalid)
-                            telemetry.error("Failed to initiate '\(self.featureName)' data upload", error: error)
                         }
+                    } catch let error {
+                        // If upload can't be initiated do not retry, so drop the batch:
+                        self.fileReader.markBatchAsRead(batch, reason: .invalid)
+                        telemetry.error("Failed to initiate '\(self.featureName)' data upload", error: error)
                     }
                 }
             } else {
                 let batchLabel = batches?.isEmpty == false ? "YES" : (isSystemReady ? "NO" : "NOT CHECKED")
                 DD.logger.debug("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(blockersForUpload.description)")
 
-                self.delay.increase()
                 self.backgroundTaskCoordinator?.endBackgroundTask()
             }
 
+            allUploadsSucceeded
```

Comment:
hmmm I think this is debatable or I do not see it well in this PR. I would only `decrease/increase` in case there was a 500 .

Reply:
That's a good call.

It should only increase the delay for those batches that `needsRetry`.